synapse/rest/admin/users.py

```diff
@@ -346,10 +354,10 @@ class UserRegisterServlet(RestServlet):
     PATTERNS = admin_patterns("/register")
     NONCE_TIMEOUT = 60
 
-    def __init__(self, hs):
+    def __init__(self, hs: "HomeServer"):
         self.auth_handler = hs.get_auth_handler()
         self.reactor = hs.get_reactor()
-        self.nonces = {}
+        self.nonces = {}  # type: Dict
```

Comment:
I think this is a mapping of strings to ints? It would be great to specify that as `Dict[str, int]`.